.github/workflows/End2EndTestApache.yml

```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   build_and_test:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```

Comment:
That is super risky... I don't think we should do that.

Reply:
Let's use the current latest instead of the latest tag

Reply:
Besides, I think I tried to bump it some time ago, and there were issues with some of the packages.

Reply:
Ok, let's try with 24.04 then.

Reply:
24.04 works for non-Python workflows. For e2e tests there is some package version collision in 24.04, but not in 22.04.

If CI passes I will go on with combination of 22 and 24 to unblock development.